src/main/java/appeng/client/render/crafting/ItemEncodedPatternBakedModel.java

```diff
@@ -227,7 +227,7 @@ public IBakedModel handleItemState( IBakedModel originalModel, ItemStack stack,
 			{
 				ItemEncodedPattern iep = (ItemEncodedPattern) stack.getItem();
 				ItemStack output = iep.getOutput( stack );
-				if( output != null )
+				if( output != null && !output.isEmpty() )
```

Comment:
A pattern should never have no outputs. The crafting system needs them to lookup the inputs, mark something as craftable etc. Being null or `ItemStack.EMPTY` might be a question. The condensed output should never contain empty for the usual reasons of not being a valid pattern, should be an empty collection not a collection of null objects and so on.

Treating `ItemEncodedPattern#getOutput()` as `@Nonnull` is certainly valid. So `ItemStack.EMPTY` is fine to use, but break on receiving `null`, because that means something like another mod created a pattern, which can certainly create more issues elsewhere.